### PR TITLE
Added more informative text when trying to install bionics depending on player's first aid skill

### PR DESCRIPTION
**Pull request by Night-Pryanik**
Players with first aid skill levels 0 and 1 will get old vague messages about taking painkillers.
Players with first aid skill level 2, 3 and 4 will get current painkiller intensity and a remark about it being less or more than half of the painkiller threshold.
Players with first aid skill level 5 and beyond (and players with MD profession) will get the exact number of painkiller intensity they need to achieve in order to make installing tolerable.

Also, players with moderate and high levels in first aid skill can realize how much time it will take for painkillers to achieve maximum effect.

Will conflict with #21433, but this PR has more priority.

### CODE REVIEW

**Comment by AlecWhite** [comment]
Given that any bottle of painkiller will tell you the maximum dose. I think it is necessary to restrict it based on skill.

**Comment by Night-Pryanik** [comment]
> Given that any bottle of painkiller will tell you the maximum dose. I think it is necessary to restrict it based on skill.

It will be rather hard to calculate maximum dose when one is taking several different painkillers.

**Comment by AlecWhite** [comment]
Why not prompt the the time when the painkillers effects will peak? It could add the painkiller effect strength and time, and then tell you if it will be enough or not at its peak.

It would need a special case for painkillers that are player made like poppy painkiller, or stuff like heroine.

**Comment by DangerNoodle** [comment]
I think this is a good idea. Characters with medical experience should be able to determine how they are affected by drugs, and this means more useful information is conveyed to the player without resorting to game-isms. In realism terms, this can be written off as someone mentally keeping track of the dosage they've taken, and working off of how much their medical experience tells them should be adequate for them.

Ideally first aid skill should allow looking at the painkiller item and getting a rough idea of how much one dose will effect them, but that can be handled in a later pull request.

**Comment by Night-Pryanik** [comment]
Added info for players with some medical experience about how much time it will take for painkillers to achieve maximum effect.

**Comment by kevingranade** [comment]
It will be clearer if you say, "it will take X minutes for the painkillers you've already taken to reach full effect.".  Right now it's unclear whether it means painkillers you've already taken or the painkillers its telling you to take.

File: src/bionics.cpp [hunk 1]
```diff
@@ -970,22 +971,42 @@ bool player::install_bionics( const itype &type, int skill_level )
...
+    }
+
     if( !has_trait( trait_id( "NOPAIN" ) ) && !has_trait( trait_id( "CENOBITE" ) ) ) {
         if( pk == 0 ) {
             popup( _( "You need to take painkillers to make installing bionics tolerable." ) );
             return false;
         } else if( pk < pain_cap / 2 ) {
-            popup( _( "You need to be a lot more numb to tolerate installing bionics.  Note that painkillers you've already taken might not be fully working yet." ) );
+            if( fa_level < 2 ) {
+                popup( _( "You need to be a lot more numb to tolerate installing bionics.  Note that painkillers you've already taken might not be fully working yet." ) );
```
Comment (kevingranade) [inline src/bionics.cpp]: Suggest changing the final clause to, "note that painkillers youve already taken could take up to an hour to achieve full effect."

**Comment by kevingranade** [comment]
More importantly though, I still don't want any raw numbers here, where did the player get those from?

**Comment by DangerNoodle** [comment]
> More importantly though, I still don't want any raw numbers here, where did the player get those from?

Game-ism because conveying information to the player that the character knows is hard.

Someone with medical background related to administering painkillers and anesthesiology should have an understanding of how much dosage is required for a given patient, the latter being an important medical field for good reason, given how much training goes into measuring and administering the correct dosage.

Conveying this in a sane manner is difficult, but it is better than not conveying information if that information is important.

**Comment by Night-Pryanik** [comment]
Well, the rationale is that people with medical skill could calculate the effect of painkillers based on their experience and info from the instructions. 
But also we already could see almost the exact effect of painkillers in the @-menu, in the speed submenu.

File: src/bionics.cpp [hunk 2]
```diff
@@ -974,23 +975,43 @@ bool player::install_bionics( const itype &type, int skill_level )
...
     if( has_trait( trait_PAINRESIST_TROGLO ) ) {
         pain_cap = pain_cap / 2;
     } else if( has_trait( trait_PAINRESIST ) ) {
         pain_cap = pain_cap / 1.5;
     }
 
+    int fa_level = get_skill_level( skill_firstaid );
+
+    if( has_trait( trait_PROF_MED ) ) {
+        fa_level =+5;
```
Comment (BevapDin) [inline src/bionics.cpp]: Why is there a `+` in front of the `5`? `fa_level = 5` is the same, but shorter (and follows the established whitespace rules).

**Comment by kevingranade** [comment]
> Well, the rationale is that people with medical skill could calculate the effect of painkillers based on their experience and info from the instructions.

This justifies the player being able to do "something", not exposing the raw numbers in the UI.  Displaying internal numbers to the player is almost always a bad idea.

This is why I suggested adding a menu that allows the player to set their dosage with a precision based on their skill, that gives the player the *ability* to control dosage without bothering them with nonsense numbers.

> But also we already could see almost the exact effect of painkillers in the @-menu, in the speed submenu.

That's a bad interface, but it already exists, so we need to replace it with something better instead of just removing it.  It existing is not a rationale for spreading the bad interface to more parts of the game.

**Comment by Night-Pryanik** [comment]
> This is why I suggested adding a menu that allows the player to set their dosage with a precision based on their skill, that gives the player the ability to control dosage without bothering them with nonsense numbers.

You mean when `a`ctivating CBM in order to install it, the query should appear that asks the player, if he want to take painkillers prior to installing? And if he answers 'yes', then the game looks through all of his painkillers in inventory, calculate the amount of them needed to achieve 100 pk intensity, and automatically consume them?

**Comment by kevingranade** [comment]
Yes, exactly that.  Not necessarally in this PR, just ditch the pinkiller numbes and this one is good to go, it's an improvement and the menu thing might take a while to put together.

**Comment by Night-Pryanik** [comment]
I left the painkillers numbers for first aid skill level 5 and beyond, because at the moment I can't think what sort of information (if not exact pk numbers) high-medical-experience users could get.

**Comment by DangerNoodle** [comment]
Estimate could be given in percentage of painkiller level required for install. Example: "Intensity of painkillers you've already taken is 85% of the threshold that will allow you to install bioncs." as a direct upgrade in information utility over the information level below it.

**Comment by kevingranade** [comment]
I agree a proportion/percentage would be an improvement.

**Pull request merged**